pennylane/templates/subroutines/commuting_evolution.py

```diff
@@ -1,148 +1,148 @@
-            partial derivative will be computed using the standard two-term shift rule applied to
-            the constituent Pauli words in the Hamiltonian individually.
-
-        shifts (tuple[int or float]): The parameter shifts to use in obtaining the
-            generalized parameter shift rules. If unspecified, equidistant shifts are used.
-
-    .. UsageDetails::
-
-        The template is used inside a qnode:
-
-        .. code-block:: python
-
-            import pennylane as qml
-
-            n_wires = 2
-            dev = qml.device('default.qubit', wires=n_wires)
-
-            coeffs = [1, -1]
-            obs = [qml.PauliX(0) @ qml.PauliY(1), qml.PauliY(0) @ qml.PauliX(1)]
-            hamiltonian = qml.Hamiltonian(coeffs, obs)
-            frequencies = (2,4)
-
-            @qml.qnode(dev)
-            def circuit(time):
-                qml.PauliX(0)
-                qml.CommutingEvolution(hamiltonian, time, frequencies)
-                return qml.expval(qml.PauliZ(0))
-
-        >>> circuit(1)
-        0.6536436208636115
-    """
-
-    num_params = 3
-    num_wires = AnyWires
-    par_domain = "R"
-    grad_method = None
-
-    def __init__(self, hamiltonian, time, frequencies=None, shifts=None, do_queue=True, id=None):
-        # pylint: disable=import-outside-toplevel
-        from pennylane.gradients.general_shift_rules import (
-            generate_shift_rule,
-        )
-
-        if not isinstance(hamiltonian, qml.Hamiltonian):
-            type_name = type(hamiltonian).__name__
-            raise TypeError(f"hamiltonian must be of type pennylane.Hamiltonian, got {type_name}")
-
-        trainable_hamiltonian = qml.math.requires_grad(hamiltonian.coeffs)
-        if frequencies is not None and not trainable_hamiltonian:
-            c, s = generate_shift_rule(frequencies, shifts)
-            recipe = qml.math.stack([c, qml.math.ones_like(c), s]).T
-            self.grad_recipe = (recipe,) + (None,) * len(hamiltonian.data)
-            self.grad_method = "A"
-
-        self.hamiltonian = hamiltonian
-        self.num_params = len(hamiltonian.data) + 1
-        self.frequencies = frequencies
-        self.shifts = shifts
-
-        super().__init__(time, *hamiltonian.data, wires=hamiltonian.wires, do_queue=do_queue, id=id)
-
-    def expand(self):
-        # uses standard PauliRot decomposition through ApproxTimeEvolution.
-        hamiltonian = qml.Hamiltonian(self.parameters[1:], self.hamiltonian.ops)
-        time = self.parameters[0]
-
-        return qml.templates.ApproxTimeEvolution(hamiltonian, time, 1).expand()
-
-    def adjoint(self):  # pylint: disable=arguments-differ
-
-        hamiltonian = qml.Hamiltonian(self.parameters[1:], self.hamiltonian.ops)
-        time = self.parameters[0]
-        frequencies = self.frequencies
-        shifts = self.shifts
-
-        return CommutingEvolution(hamiltonian, -time, frequencies, shifts)
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
```

Comment:
@dime10 it looks like this file has been entirely modified?

Reply:
If this was unintentional, best to revert any changes to this file via
```python
git checkout master pennylane/templates/subroutines/commuting_evolution.py
```

Reply:
Oh boy, some weird git stuff going on 😅 It's not part of any of my commits. Apparently GitHub says it was introduced in a merge from master.